va/va.go

```diff
@@ -464,32 +463,37 @@ func (va *ValidationAuthorityImpl) validateDNS01(ctx context.Context, identifier
 }
 
 func (va *ValidationAuthorityImpl) checkCAA(ctx context.Context, identifier core.AcmeIdentifier) *probs.ProblemDetails {
-	// Check CAA records for the requested identifier
-	present, valid, err := va.checkCAARecords(ctx, identifier)
-	if err != nil {
-		return bdns.ProblemDetailsFromDNSError(err)
+	var prob *probs.ProblemDetails
+	if va.caaClient != nil {
+		prob = va.checkCAAService(ctx, identifier)
+	} else {
+		present, valid, err := va.checkCAARecords(ctx, identifier)
+		if err != nil {
+			prob = bdns.ProblemDetailsFromDNSError(err)
+		} else {
+			// AUDIT[ Certificate Requests ] 11917fa4-10ef-4e0d-9105-bacbe7836a3c
+			va.log.AuditInfo(fmt.Sprintf(
+				"Checked CAA records for %s, [Present: %t, Valid for issuance: %t]",
+				identifier.Value,
+				present,
+				valid,
+			))
+			if !valid {
```

Comment:
This level of nesting is getting a little deep to follow. I think it might make it clearer to split this into checkCAA and checkCAAWithFallback, so the `caaClient` case can do a short-circuit return and the rest can be outdented a bunch. What do you think?